src/wr_control_drive_science/src/TeleopScienceControl.cpp

```diff
@@ -0,0 +1,154 @@
+#include "ros/assert.h"
+#include "ros/ros.h"
+#include "std_msgs/Float32.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/UInt32.h"
+#include "std_msgs/Float64.h"
+#include "std_msgs/Int16.h"
+#include "ros/timer.h"
+#include <fcntl.h>
+#include <termios.h>
+#include <string>
+#include <array>
+
+constexpr std::uint32_t MESSAGE_CACHE_SIZE = 10;
+constexpr float TIMER_CALLBACK_DURATION = 5;
+
+ros::Publisher moistureSensor1;
+ros::Publisher moistureSensor2;
+ros::Publisher moistureSensor3;
+int baudRate;
+std::string file;
+int fd;
+
+void moistureCallback(const ros::TimerEvent &timerEvent) {
+    std::array<float, 3> buf{0, 0, 0};
```

Comment:
3, in this context, is a magic number.  There should be some `constexpr` variable or ROSParam for how many samples per reading we expect